packages/itwin/measure-tools/src/measurements/PerpendicularDistanceMeasurement.ts

```diff
@@ -0,0 +1,285 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { Point3d, XYZProps } from "@itwin/core-geometry";
+import { ColorDef, LinePixels } from "@itwin/core-common";
+import type { DecorateContext } from "@itwin/core-frontend";
+import { GraphicType, IModelApp, QuantityType } from "@itwin/core-frontend";
+import { FormatterUtils } from "../api/FormatterUtils";
+import { Measurement, type MeasurementEqualityOptions, MeasurementSerializer, type MeasurementWidgetData } from "../api/Measurement";
+import { MeasurementPropertyHelper } from "../api/MeasurementPropertyHelper";
+import { MeasureTools } from "../MeasureTools";
+import { DistanceMeasurement } from "./DistanceMeasurement";
+import type { MeasurementProps } from "../api/MeasurementProps";
+import { MeasurementPreferences, MeasurementPreferencesProperty } from "../api/MeasurementPreferences";
+import { MeasurementManager } from "../api/MeasurementManager";
+
+export enum PerpendicularMeasurementType {
+  Width = "width",
+  Height = "height",
+}
+
+/**
+ * Props for serializing a [[PerpendicularDistanceMeasurement]].
+ */
+export interface PerpendicularDistanceMeasurementProps extends MeasurementProps {
```

Comment:
This props interface should carry `secondaryLine?: XYZProps[]`. Consider the case when you serialize then deserialize the measurement. It will not have that data and will end up looking differently.

Reply:
Moved getHeightPoints to PerpendicularDistanceMeasurement.ts